### PR TITLE
api refactor + remove duplicates

### PR DESCRIPTION
**Pull request by Ali-Razmjoo**
not ready for review yet...

### CODE REVIEW

**Comment by Ali-Razmjoo** [comment]
this pr and #122 must merge together and sync up then update all APIs in webui and update docs in #177

File: database/connector.py [hunk 1]
```diff
@@ -246,13 +246,13 @@ def insert_to_events_data_collection(event_data: EventData):
     Returns:
         inserted_id
     """
-    event_data.machine_name = \
-        network_config["real_machine_identifier_name"]
+    event_data.machine_name = network_config["real_machine_identifier_name"]
```
Comment (dhirensr) [inline database/connector.py:249]: name should be changed here too right if we are changing above?

**Comment by Ali-Razmjoo** [comment]
@dhirensr ui should be fix with #122 , it's not working now and it will have conflict with that pr, i will fix it tomorrow ish

File: database/connector.py (hunk 1 above)
Comment (Ali-Razmjoo) [reply database/connector.py:248]: right, fixed.

**Comment by Ali-Razmjoo** [comment]
backend is almost good (I think it's a bit buggy when submitting honeypot events we should check that) and I refactor our API, now we need to sync it with frontend which is our worse code now and absolutely needs refactoring, I am also open to deface the whole thing and build it from scratch.

File: api/server.py [hunk 2]
```diff
@@ -227,1092 +240,197 @@ def get_static_files(path):
...
     except Exception:
-        return flask_null_array_response()
+        return abort(404)
 
 
-@app.route("/api/events/module-names", methods=["GET"])
+@app.route("/api/core/list/modules", methods=["GET"])
 def all_module_names():
     """
     Get top passwords used according to module
```
Comment (dhirensr) [inline api/server.py:423]: doc string of this also needs to be changed

File: api/server.py [hunk 3]
```diff
@@ -227,1092 +240,197 @@ def get_static_files(path):
...
+        abort(500)
 
 
-@app.route("/api/events/network-events-ips", methods=["GET"])
-def top_ten_ips_in_network_events():
+@app.route("/api/events/count/groupby/<event_type>/<element>", methods=["GET"])
+def groupby_element(event_type, element):
     """
-    get top ten repeated ips in network events
+    get top ten repeated ips in honeypot events
```
[the remark this answers is not in the document]
Comment (ChakshuGupta) [reply api/server.py:282]: I'll take a look.

File: api/server.py (hunk 2 above)
Comment (ChakshuGupta) [reply api/server.py:423]: ok

**Comment by ChakshuGupta** [comment]
@dhirensr I faced errors too but I thought I fixed it in the latest commits. Did you pull in the changes when testing?

**Comment by Ali-Razmjoo** [comment]
I've removed redundant function for exploring pcap, now they can be considered as events, and they can be downloaded by `md5` instead of `_id`.

```
batman@batmans-MacBook-Pro ~/Documents/GitHub/OWASP-Honeypot (enhancements) $ curl http://localhost:5000/api/events/count/pcap
{"count":1,"date":null}
batman@batmans-MacBook-Pro ~/Documents/GitHub/OWASP-Honeypot (enhancements) $ curl http://localhost:5000/api/events/count/groupby/pcap/machine_name
[{"count":1,"machine_name":"stockholm_server_1"}]
batman@batmans-MacBook-Pro ~/Documents/GitHub/OWASP-Honeypot (enhancements) $ curl http://localhost:5000/api/events/explore/pcap
[{"chunkSize":261120,"date":"2020-08-16 23:52:58","filename":"captured-traffic-1597614778.pcap","length":286720,"machine_name":"stockholm_server_1","md5":"282e14c5b89ff2af63f4146fbd0a6c68","splitTimeout":3600,"uploadDate":"Sun, 16 Aug 2020 21:53:15 GMT"}]
batman@batmans-MacBook-Pro ~/Documents/GitHub/OWASP-Honeypot (enhancements) $ curl http://localhost:5000/api/pcap/download?md5=282e14c5b89ff2af63f4146fbd0a6c68 > f.pcap
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  280k    0  280k    0     0  8484k      0 --:--:-- --:--:-- --:--:-- 8484k
batman@batmans-MacBook-Pro ~/Documents/GitHub/OWASP-Honeypot (enhancements) $ ls -lah f.pcap 
-rw-r--r--  1 batman  staff   280K Aug 17 22:28 f.pcap

```

@ChakshuGupta can you please modify the webui? 
@dhirensr it might be something you want to add to docs.

**Comment by ChakshuGupta** [comment]
@Ali-Razmjoo Yeah I am working on it.

**Comment by dhirensr** [comment]
when I don't have any pcap files and I hit http://127.0.0.1:5000/api/pcap/download , I am getting this error on the terminal 
```
127.0.0.1 - - [18/Aug/2020 16:08:34] "GET /api/events/explore/pcap?datatable_id=%23file-datatable&date=2020-06-01%7C2020-08-18&limit=1000&_=1597759618650 HTTP/1.1" 200 -
Exception ignored in: <bound method Cursor.__del__ of <gridfs.grid_file.GridOutCursor object at 0x7f799d3ad128>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pymongo/cursor.py", line 241, in __del__
    self.__die()
  File "/usr/local/lib/python3.6/dist-packages/pymongo/cursor.py", line 298, in __die
    already_killed = self.__killed
AttributeError: 'GridOutCursor' object has no attribute '_Cursor__killed'


```

**Comment by ChakshuGupta** [comment]
@dhirensr I forgot to add a check there. WIll add it.

**Comment by ChakshuGupta** [comment]
@dhirensr I added the check in the UI

**Comment by dhirensr** [comment]
@ChakshuGupta : I'm facing a problem with the log explorer, when I selected network events the modules selection is getting locked and all modules is getting selected as default. in all other events it's working fine.

**Comment by Ali-Razmjoo** [comment]
I've merge log explorer for pcap file...

anyway now we need to test everything before merge, also i noticed honeypot_events of mine is empty, maybe some bug in core/network.py?

**Comment by Ali-Razmjoo** [comment]
i had something wrong with my config.py.... honeypot doens't have issue, anyway I've some crashes on ssh/strong_password, that needs to be tested.
